zero/src/lib.rs

```diff
@@ -29,3 +33,30 @@ pub fn version() -> String {
     let kernel_hash = &**prover_state::persistence::KERNEL_HASH;
     format!("{pkg_name} ({git_describe}) (kernel hash: {kernel_hash}) [built: {timestamp}]")
 }
+
+/// Like `#[serde(with = "::hex")]`, but tolerates and emits leading `0x`
+/// prefixes
+mod hex {
```

Comment:
I feel there is a lot of potential for some `zero/src/common` folder/submodule.  Looking at the moment at the 20 files (submodules) in the zero crate it is not obvious what is important, what is core of the `zero` logic and purpose. I think we have went too far in the "un-cratezation".